### PR TITLE
add gtfs aggregator checks

### PR DESCRIPTION
**Pull request by owades**
# Description

Use the new aggregator scraper to check for each feed's presence on transit.land and mobility database.

One part of this that I'd like feedback on is the way that I ignore HTTP vs HTTPS (removing everything before "://" in the URL). Specifically I see that some aggregators have URLs listed starting with "ftp://" and I'm not sure whether these should be included or omitted.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation
- [ ] agencies.yml

## How has this been tested?
Manually reviewed multiple passing and failing values. As an example of a failing value, I found that the URL we have listed for Desert Roadrunner is `https://data.trilliumtransit.com/gtfs/paloverde_valley-ca-us/paloverde_valley-ca-us.zip`, while the URL listed on transit.land is `http://data.trilliumtransit.com/gtfs/paloverde-valley-ca-us/paloverde-valley-ca-us.zip` (note the underscore/hyphen discrepancy between "paloverde" and "valley" - this is what the check is for!)

Also ran dbt build on both tables and confirmed no errors, and no new downstream errors

### CODE REVIEW

File: warehouse/models/intermediate/gtfs/gtfs_quality/int_gtfs_quality__schedule_feed_guideline_index_aggregator.sql [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+WITH int_gtfs_quality__schedule_feed_guideline_index AS (
+    SELECT * FROM {{ ref('int_gtfs_quality__schedule_feed_guideline_index') }}
+    -- 1/13/23 is when the aggregator scraper started running
+    WHERE date >= '2023-01-13'
+),
+
+distinct_aggregators AS (
+    SELECT DISTINCT aggregator
+      FROM {{ ref('stg_gtfs_quality__scraped_urls') }}
+),
```
Comment (atvaccaro) [inline warehouse/models/intermediate/gtfs/gtfs_quality/int_gtfs_quality__schedule_feed_guideline_index_aggregator.sql:12]: nit: I think you could pull the minimum date from stg_gtfs_quality__scraped_urls and use that to filter the index

File: warehouse/models/intermediate/gtfs/gtfs_quality/int_gtfs_quality__feed_aggregator_rt.sql [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+           RIGHT(feed_url_str,LENGTH(feed_url_str) - STRPOS(feed_url_str, "://") - 2) AS url_no_scheme
+    FROM {{ ref('stg_gtfs_quality__scraped_urls') }}
+),
+
+
+dim_gtfs_datasets AS (
+    SELECT *,
+           RIGHT(uri,LENGTH(uri) - STRPOS(uri, "://") - 2) AS url_no_scheme
+    FROM {{ ref('dim_gtfs_datasets') }}
+),
```
Comment (atvaccaro) [inline warehouse/models/intermediate/gtfs/gtfs_quality/int_gtfs_quality__feed_aggregator_rt.sql:16]: nit: this feels a bit fragile, regex may be a bit more explicit?

File: warehouse/models/intermediate/gtfs/gtfs_quality/int_gtfs_quality__feed_aggregator_rt.sql [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+             WHEN t1.aggregator = 'transitland' AND t1.feed_type = "service_alerts" THEN {{ service_alerts_feed_on_transitland() }}
+             WHEN t1.aggregator = 'mobility_database' AND t1.feed_type = "vehicle_positions" THEN {{ vehicle_positions_feed_on_mobility_database() }}
+             WHEN t1.aggregator = 'mobility_database' AND t1.feed_type = "trip_updates" THEN {{ trip_updates_feed_on_mobility_database() }}
+             WHEN t1.aggregator = 'mobility_database' AND t1.feed_type = "service_alerts" THEN {{ service_alerts_feed_on_mobility_database() }}
+             END AS check,
+        {{ feed_aggregator_availability_rt() }} AS feature,
+        CASE
+            WHEN t3.aggregator IS NOT null THEN "PASS"
+            ELSE "FAIL"
+        END AS status,
```
Comment (atvaccaro) [inline warehouse/models/intermediate/gtfs/gtfs_quality/int_gtfs_quality__feed_aggregator_rt.sql:42]: nit: you could nest the `case` statements to reduce repetition

File: warehouse/models/intermediate/gtfs/gtfs_quality/int_gtfs_quality__feed_aggregator_schedule.sql [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+),
+
+dim_schedule_feeds AS (
+    SELECT * FROM {{ ref('dim_schedule_feeds') }}
+),
+
+dim_gtfs_datasets AS (
+    SELECT *,
+           RIGHT(uri,LENGTH(uri) - STRPOS(uri, "://") - 2) AS url_no_scheme
+    FROM {{ ref('dim_gtfs_datasets') }}
```
Comment (atvaccaro) [inline warehouse/models/intermediate/gtfs/gtfs_quality/int_gtfs_quality__feed_aggregator_schedule.sql:18]: I'd recommend using a macro if this is being done in multiple places

**Review by atvaccaro — CHANGES_REQUESTED** [review]
One broad comment: I _might_ recommend only doing the URL protocol handling in the joins, and maybe dig more into the ftp records? IMO I think that if we have HTTP(S) and the aggregator has FTP that's probably something we want to flag? Can discuss more in Slack/Zoom if you'd like.

**Comment by owades** [comment]
> One broad comment: I _might_ recommend only doing the URL protocol handling in the joins, and maybe dig more into the ftp records? IMO I think that if we have HTTP(S) and the aggregator has FTP that's probably something we want to flag? Can discuss more in Slack/Zoom if you'd like.

Ok, a bit more info about the ftp thing. There are only 7 URLs listed (if they are even technically URLs) starting with "ftp://". They're all from transitland, and only 2 appear to be in the US.

For at least [this one](https://www.transit.land/feeds/f-9qbdx-santarosacitybus), it's listed as the url from "historic GTFS". Either way, I feel good about filtering these out. Going to do so now.

```
ftp.scgov.net/pub/scat/SCATGTFS.zip
ftp.ci.santa-rosa.ca.us/SantaRosaCityBus/google_transit.zip
gtfs.mot.gov.il/israel-public-transportation.zip
ftp.geo.euskadi.net/cartografia/Transporte/Moveuskadi/ATTG/dbus/google_transit.zip
ftp.geo.euskadi.net/cartografia/Transporte/Moveuskadi/MetroBilbao/google_transit.zip
ftp.geo.euskadi.net/cartografia/Transporte/Moveuskadi/Euskotren/google_transit.zip
ftp.geo.euskadi.net/cartografia/Transporte/Moveuskadi/FunicularArtxanda/google_transit.zip
```

**Comment by owades** [comment]
@atvaccaro thanks for pushing me to write higher quality stuff. I've made all of the relevant updates. I'm also no longer filtering out the ftp addresses, as the regex logic in the join does it automatically (by requiring text to contain "http"

**Review by atvaccaro — APPROVED** [review]
lgtm!

**Pull request merged**